js/NavMainComponent.js

```diff
@@ -100,6 +181,92 @@ NavMainComponent.prototype.closeSubNavs = function() {
     var component = this;
 
     component.$html.find('.nav-secondary .nav-list').removeClass('is-active');
+  
+};
+
+/* Detect window height, adjust the number of items in the primary nav and check when to add "More" option */
+
+/* Notes:
+- Feels like the JS should handle the creation of the third level menu to avoid unlessisary markup changes
+- no need for this to be on prototype
+*/
+
+NavMainComponent.prototype.adjustNavItems = function() {
+
+    var component = this,
+        availableHeight = component.$window.height(),
+        navItemsHeight = (component.$html.find('.nav-primary .nav-items').outerHeight(true) + component.$html.find('.jadu-branding').outerHeight(true)),
+        moreIconHeight = 72, // Pre calculated height of the "More" nav item
+        navItemsCountTotal = component.$html.find('.nav-primary .nav-items li').length,
+        i = 2, // This number represents the item before the last in the nth-last-child
+        numberOfHiddenNavItems = 0;
+
+    if (navItemsHeight + moreIconHeight > availableHeight) {
+        component.hidePrimaryNavItems(navItemsHeight, moreIconHeight, availableHeight, i);
+        numberOfHiddenNavItems = component.$html.find('.nav-primary .nav-items li:hidden').length;
+        component.addMoreNavItem(numberOfHiddenNavItems);
+        component.hideMoreCategoriesTopItems(navItemsCountTotal, numberOfHiddenNavItems);
+    } else {
+        // Unhide items if they were hidden and there is space in the primary nav
```

Comment:
More comments never harmed anyone. Added a few of them to increase human understanding if we revisit the code